### PR TITLE
dwl: init at 0.1

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/dwl/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  buildInputs = [
+    libxcb
+    libxkbcommon
+    wayland
+    wayland-protocols
+    wlroots
+  ] ++ lib.optionals enable-xwayland [ xwayland libX11 ];
+
+  # Allow users to set their own list of patches
+  inherit patches;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/dwl/default.nix:37]: Why do we need this? This can be easily done with an overlay.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112509` at e8f0c64a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/eac836e94770517021a2db6dfb64e150">dwl</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112509` at e8f0c64a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/ba5943699190d503f02c5ed3170de4f1">dwl</a></li></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112509` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>dwl: log <a href="https://termbin.com/5lae">https://termbin.com/5lae</a></li>
  </ul>
</details>

File: pkgs/applications/window-managers/dwl/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/window-managers/dwl/default.nix:37]: dwl follows this strange tradition of "customize by patching the source code". Indeed they have a [dedicated page listing some patches](https://github.com/djpohly/dwl/wiki/Patches).

Comment (SuperSandro2000) [reply pkgs/applications/window-managers/dwl/default.nix:37]: I am not a fan of that practice. Not sure what the best way is here.

**Pull request merged**